src/preact/hooks/useLocalStorage.js

```diff
@@ -0,0 +1,29 @@
+import {useCallback, useRef, useState} from '#preact';
+
+// eslint-disable-next-line local/no-forbidden-terms, local/no-global
+const storage = window.localStorage;
+
+/**
+ * @param {string} key
+ * @param {any} [defaultValue]
+ * @return {{ "0": any, "1": function(any): void }}
+ */
+export function useLocalStorage(key, defaultValue) {
+  // Keep track of the state locally:
+  const [value, setValue] = useState(() => {
+    const json = storage.getItem(key);
```

Comment:
`storage` can be null in private mode, and it'll always be null during SSR. We need to guard on nullish values.

Reply:
Added null guards and  `try` blocks to ensure safe access